### DISCUSSION TITLE
Complex nested license expression example from docs

### DISCUSSION THREAD

**Discussion by joseph-grabinger** [Homebrew/discussions#5321] [Everyday usage] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I recently found the example license expression listed in https://docs.brew.sh/License-Guidelines#complex-spdx-license-expressions. 

The example given is the following:
 ```
license any_of: [
  "MIT",
  :public_domain,
  { all_of: ["0BSD", "Zlib", "Artistic-1.0+"],
  "Apache-2.0" => { with: "LLVM-exception" } },
]
```

Since the brew JSON API outputs the license in natural language using "and" & "or", I was wondering what the given example expression would resolve to? 

I could imagine either of the two: 

1. `MIT and Public Domain and (0BSD or Zlib or Artistic-1.0+ or (Apache-2.0 with LLVM-exception))`
2. `MIT and Public Domain and (0BSD or Zlib or Artistic-1.0+) and (Apache-2.0 with LLVM-exception)`

**Comment by p-linnane** [comment, marked as answer]
The intent for this one would be `MIT or Public Domain or (0BSD and Zlib and Artistic-1.0+) or (Apache-2.0 with LLVM-exception`
